e2e/tests/i18n/settings.spec.ts

```diff
@@ -0,0 +1,187 @@
+  }) => {
+    /**
+     * Get to the settings page
+     */
+    await page.getByRole('link', { name: 'Settings' }).click();
+    await page.getByRole('link', { name: 'Internationalization' }).click();
+
+    /**
+     * Assert that the page has the expected elements & data, the locales installed in the test-app.
+     */
+    await expect(page.getByRole('heading', { name: 'Internationalization' })).toBeVisible();
+    expect(await page.getByRole('row').all()).toHaveLength(5);
+    for (const locale of LOCALES) {
+      expect(page.getByRole('gridcell', { name: locale, exact: true })).toBeVisible();
+    }
+
+    /**
+     * Add a new locale to the current list.
+     */
+    await page.getByRole('button', { name: 'Add new locale' }).click();
+    await expect(page.getByRole('dialog', { name: 'Add new locale' })).toBeVisible();
+    expect(page.getByRole('heading', { name: 'Configuration' })).toBeVisible();
+    await page.getByRole('combobox', { name: 'Locales' }).click();
+    await page.press('body', 'I');
+    await page.getByRole('option', { name: 'Italian (it)' }).click();
+    await expect(page.getByRole('button', { name: 'Save' })).toBeEnabled();
+    await page.getByRole('button', { name: 'Save' }).click();
+    await expect(page.getByText('Success:Locale successfully added')).toBeVisible();
+
+    /**
+     * Next, we'll navigate to our shop single type & add the a localised version of this document.
+     */
+    await page.getByRole('link', { name: 'Content Manager' }).click();
+    await page.getByRole('link', { name: 'Shop' }).click();
+    await page.waitForURL(/\/admin\/content-manager\/single-types\/api::shop.shop(\?.*)?/);
+    await expect(page.getByRole('heading', { name: 'UK Shop' })).toBeVisible();
+    await page.getByRole('combobox', { name: 'Locales' }).click();
+    expect(await page.getByRole('option').all()).toHaveLength(5);
+    for (const locale of [...LOCALES, 'Italian (it)']) {
+      await expect(page.getByRole('option', { name: locale })).toBeVisible();
+    }
+    await page.getByRole('option', { name: 'Italian (it)' }).click();
+    await expect(page.getByRole('heading', { name: 'Untitled' })).toBeVisible();
+
+    /**
+     * Because this is technically a new entry, we should assert that the document actions are disabled.
+     */
+    await expect(page.getByRole('button', { name: 'Publish' })).toBeDisabled();
+    await expect(page.getByRole('button', { name: 'Save' })).toBeDisabled();
+    await expect(page.getByRole('button', { name: 'More document actions' })).toBeDisabled();
+    await expect(page.getByRole('button', { name: 'More actions' })).toBeEnabled();
+    await page.getByRole('button', { name: 'More actions' }).click();
+    await expect(page.getByRole('menuitem', { name: 'Edit the model' })).toBeEnabled();
+    await expect(page.getByRole('menuitem', { name: 'Configure the view' })).toBeEnabled();
+    await expect(page.getByRole('menuitem', { name: 'Delete locale' })).toBeDisabled();
```

Comment:
I wonder if 'Delete locale' in the CM could be confusing and make people think
they're removing the locale completely?

Reply:
Hey @jhoward1994 it could, indeed. I'm gonna speak about this with Paridhi :)